pkgs/top-level/all-packages.nix

```diff
@@ -10904,15 +10904,16 @@ in
   };
 
   # This is for e.g. LLVM libraries on linux.
-  gccForLibs =
+  gccForLibs = if (stdenv.targetPlatform.isLinux && !(stdenv.targetPlatform.useLLVM or false)) then
     # with gcc-7: undefined reference to `__divmoddi4'
     if stdenv.targetPlatform.isi686
       then gcc6.cc
     else if stdenv.targetPlatform == stdenv.hostPlatform && targetPackages.stdenv.cc.isGNU
     # Can only do this is in the native case, otherwise we might get infinite
     # recursion if `targetPackages.stdenv.cc.cc` itself uses `gccForLibs`.
       then targetPackages.stdenv.cc.cc
-    else gcc.cc;
+    else gcc.cc
+  else null;
```

Comment:
I think this is causing the eval error. The package should never be null.

Reply:
Got it, that's tricky. Would the idiomatic thing be to mark gcc as broken and then check for that rather than its nullness?

Reply:
Marking something broken based on targetPlatform sounds weird to me.

Reply:
Well, as far as I know, that is what's going on - gcc is broken on macOS. Or at least, that's what I thought back in October...